pkg/util/containers/v2/metrics/provider/cache.go

```diff
@@ -0,0 +1,71 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2021-present Datadog, Inc.
+
+package provider
+
+import (
+	"sync"
+	"time"
+)
+
+type cacheEntry struct {
+	value     interface{}
+	err       error
+	timestamp time.Time
+}
+
+// Cache provides a caching mechanism based on staleness toleration provided by requestor
+type Cache struct {
+	cache       map[string]cacheEntry
+	cacheLock   sync.RWMutex
+	gcInterval  time.Duration
+	gcTimestamp time.Time
+}
+
+// NewCache returns a new cache dedicated to a collector
+func NewCache(gcInterval time.Duration) *Cache {
```

Comment:
I don't particularly like `pkg/util/cache.Cache` either, but would it make sense to use it here?